apps/model_eval_viz/app.py

```diff
@@ -98,21 +98,58 @@ def _get_data(_filepath:str)->gpd.GeoDataFrame:
     options=state_list,
 )
 
-# @pn.depends(state=' '.join(state_selector.param.value))
+
+base_map_options = {
+    'OpenStreetMap': gv.tile_sources.OSM,
+    'ESRI Imagery': gv.tile_sources.EsriImagery,
+    'ESRI World Street Map': gv.tile_sources.EsriWorldStreetMap,
+}
+
+map_selector = pn.widgets.Select(
+    description="Use to select Base Map",
+    name="Select a Base Map",
+    options=list(base_map_options.keys()),
+
+)
+
+
+def display_map(map:list=map_selector.value)->gv.tile_sources:
+    '''
+    Display a map from a selected tile source or an initial extent.
+
+    Parameters:
+        map (list, optional): List of tile sources to select from. Defaults to `map_selector.value`.
+
+    Returns:
+        gv.tile_sources: A Tile source from the GeoViews library.
+    '''
+
+    if len(map) > 0:
+        map = base_map_options[map_selector.value]
+    else:
+        #TODO find initial extent states.geometry.total_bounds
+        map = gv.tile_sources.OSM
+    return map
+
+# create a DynamicMap to allow Panel to link map_selector with a Geoviews(Holoviews under the hood) object
+displayed_map = hv.DynamicMap(pn.bind(display_map, map=map_selector))
```

Comment:
Nothing to change here, just a small explanation.

Setting `value` in the `map_selector` earlier will automatically pass that to this function.